### PR TITLE
[FLINK-30336][BP 3.0] Sync missing e2e tests 

### PR DESCRIPTION
**Pull request by MartijnVisser**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,6 +32,10 @@ on:
         required: true
         type: boolean
 
+concurrency:
+  group: pr-${{ github.ref }}
```
Comment (zentol) [inline .github/workflows/ci.yml:36]: This should be limited to PRs.

Comment (zentol) [reply .github/workflows/ci.yml:36]: It should also be scoped by workflow so we don't cancel a weekly build on main because a commit was pushed.

Comment (MartijnVisser) [reply .github/workflows/ci.yml:36]: Based on https://stackoverflow.com/questions/70928424/limit-github-action-workflow-concurrency-on-push-and-pull-request I would guess we should make it `group: ${{ github.head_ref || github.ref_name }}` and move this to `push_pr.yml`. 

I'll update the PR

File: .github/workflows/push_pr.yml [hunk 2]
```diff
@@ -18,6 +18,11 @@
 
 name: Build flink-connector-pulsar
 on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.head_ref || github.ref_name }}
```
Comment (zentol) [inline .github/workflows/push_pr.yml:23]: Probably a good idea to include `github.workflow` as well, if we ever add more workflows for PRs/commits.

File: flink-connector-pulsar-e2e-tests/pom.xml [hunk 3]
```diff
@@ -77,6 +73,49 @@ under the License.
...
+								<goals>
+									<goal>test</goal>
+								</goals>
+								<configuration>
+									<includes>
+										<include>**/*.*</include>
+									</includes>
+									<excludedGroups>${excludeE2E}</excludedGroups>
+									<!-- E2E tests must not access flink-dist concurrently. -->
+									<forkCount>1</forkCount>
```
Comment (zentol) [inline flink-connector-pulsar-e2e-tests/pom.xml:95]: this shouldn't matter nowadays

File: pom.xml [hunk 4]
```diff
@@ -45,10 +45,12 @@ under the License.
     <modules>
         <module>flink-connector-pulsar</module>
         <module>flink-sql-connector-pulsar</module>
+        <module>flink-connector-pulsar-e2e-tests</module>
     </modules>
 
     <properties>
         <flink.version>1.16.0</flink.version>
+        <pulsar.version>2.10.1</pulsar.version>
```
Comment (zentol) [inline pom.xml:53]: should this be de-duplicated with the version in `flink-connector-pulsar`?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MartijnVisser) [reply .github/workflows/ci.yml]: I've moved it out of this PR for now, created https://issues.apache.org/jira/browse/FLINK-30353 as follow up

File: flink-connector-pulsar-e2e-tests/pom.xml (hunk 3 above)
Comment (MartijnVisser) [reply flink-connector-pulsar-e2e-tests/pom.xml]: Fixed

File: pom.xml (hunk 4 above)
Comment (MartijnVisser) [reply pom.xml:53]: Yes missed it, fixed now

**Comment by MartijnVisser** [comment]
@syhily Could you review this?

**Review by zentol — APPROVED** [review]

**Pull request merged**